### PR TITLE
Autoinstall

### PR DESCRIPTION
**Pull request by josenavas**
Uses the new QiitaPlugin class for installing the plugin.

Am I the only one who feels great about removing duplicated code and magic numbers?

### CODE REVIEW

**Comment by antgonza** [comment]
Looks really good, a couple of comments:
- [ ] How do you add more db references and define which pipeline can use them? My question is because we currently have 3 DBs: GG, UNITE, & Silva, and we are going to be adding more for shotgun and we don't want those pipelines to get "confuse" between them
- [ ] Can you add a coverage file to the repo? Something like [this](https://github.com/qiita-spots/qp-shotgun/blob/master/.coveragerc).

File: qp_target_gene/__init__.py [hunk 1]
```diff
@@ -5,3 +5,152 @@
...
+plugin.register_command(sl_cmd)
+
+# Define the Split libraries FASTQ command
+req_params = {'input_data': ('artifact', ['FASTQ', 'per_sample_FASTQ'])}
+opt_params = {
+    'barcode_type': ['string', 'golay_12'],
+    'max_bad_run_length': ['integer', '3'],
+    'max_barcode_errors': ['float', '1.5'],
+    'min_per_read_length_fraction': ['float', '0.75'],
+    'phred_offset': ['string', ''],
```
Comment (antgonza) [inline qp_target_gene/__init__.py]: This one is really an option: auto, 33, 64, right?

Comment (josenavas) [inline qp_target_gene/__init__.py]: Changed to choice - this change is pairing https://github.com/biocore/qiita/pull/1963 which I  just noticed that have errors and I'm fixing right now.

**Comment by josenavas** [comment]
I've added coverage - the comment about the database I think we discussed it offline and it looks like we need some changes in main Qiita to support that correctly.

**Pull request merged**